stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -184,6 +184,59 @@ function eigen!(A::AbstractMatrix, C::Cholesky; sortby::Union{Function,Nothing}=
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
 
+# Bunch-Kaufmann (LDLT) based solution for generalized eigenvalues and eigenvectors
+function eigen(A::AbstractMatrix{T}, B::BunchKaufman{T,<:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    eigen!(copy(A), copy(B); sortby)
+end
+function eigen!(A::AbstractMatrix{T}, B::BunchKaufman{T,<:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    D = copy(B.D)
+    M = B.uplo == 'U' ? B.U : B.L
+    A .= A[B.p, B.p]
```

Comment:
This way, the rhs will first be allocated, and then copied to `A`. Can't we use the `apply_*` functions that I referenced, in order to do it completely allocation-free?

Reply:
Thank you for this comment. Please see my post below for my response.